cli/command/swarm/init.go

```diff
@@ -44,6 +45,7 @@ func newInitCommand(dockerCli command.Cli) *cobra.Command {
 	flags.BoolVar(&opts.forceNewCluster, "force-new-cluster", false, "Force create a new cluster from current state")
 	flags.BoolVar(&opts.autolock, flagAutolock, false, "Enable manager autolocking (requiring an unlock key to start a stopped manager)")
 	flags.StringVar(&opts.availability, flagAvailability, "active", `Availability of the node ("active"|"pause"|"drain")`)
+	flags.StringVar(&opts.defaultAddrPool, flagDefaultAddrPool, "", "List of default subnet addresses followed by subnet size (format: <subnet,subnet,..:subnet-size)")
```

Comment:
This flag likely needs an API version annotation, so that it's hidden on daemons not having this feature

Reply:
A bit unsure about the format used for this option as well (combination of comma-separated and colon-separated options); also wondering; must all address pools have the same subnet size? I can foresee situations where sizes can differ for each (could use input on that one)

Reply:
We(our networking team) had gone through multiple rounds of discussion as part of design and decided to support single subnet size for all subnets.  Main reason is , currently there is no CLI option to explicitly specify  what kind of subnet (ex /24 or /20) user want when they create network.

Reply:
@thaJeztah I didn't understand how I should address below comment. I will sync with you offline
"This flag likely needs an API version annotation, so that it's hidden on daemons not having this feature"

Reply:
for the annotation, see for example https://github.com/docker/cli/blob/48fbb12b7ca2b6c1536baa2a3cbedfb4ebf8eb53/cli/command/stack/deploy.go#L62

the API version in that annotation will enable the flag if the daemon supports that version of the API; if not (if the daemon is older), the flag is disabled and hidden

Reply:
For the format used to specify the address pools, I was thinking to use the same format as is used for the daemon option; 
https://github.com/moby/moby/pull/36396

The API can use the same format as is used in the daemon.json;

```json
{
  DefaultAddrPools:[{"base":"172.80.0.0/16","size":24},{"base":"172.90.0.0/16","size":24}]
}
```

Reply:
That was initial design. But after detailed discussion on supporting different subnet size for each subnet and going through the complexity of handling them in libnetwork, we decided to support only one subnet size. Hence removed previous format and made it simple.  If the user doesn't specify subnet size, default value 24 will be taken.  Its not mandatory field.

Reply:
So the daemon option we have for local subnets no longer works?

And design is all fixed, no review here, just press merge?

Reply:
local scope still works(Daemon option).  This is for global scope and work (swarm init option). So its kind of two different feature now.

We have design document which I can share it with you. Pls let me know.

Reply:
I thought I responded to this question. May be in libnetwork PR.   Local scope configuration works in Daemon config and global scope configuration works in swarm Init . Now its kind of two separate feature.

Reply:
@thaJeztah added API version info

Reply:
It might be cleaner to have a separate `default-addr-pool-size` parameter, which could default to 24.   `:` is the group separator for IPv6 literals, so it is confusing also to use it to separate fields in a parameter.